lib/three_scale/middleware/multitenant.rb

```diff
@@ -85,10 +85,10 @@ def find_cookie_store(app)
       end
 
       module EnforceTenant
-        # adding after_initialize does not work, it would have to be done for every model
-        def after_initialize
-          enforce_tenant!
-          super if defined?(super)
+        extend ActiveSupport::Concern
+
+        included do
+          after_initialize :enforce_tenant!
```

Comment:
Much better

Reply:
This seems to be the way to hook logic after AR object initialization. To be 100% equivalent is probably not. At the very least this syntax should allow registering multiple hooks while previously it was just a method. Any specific concerns that you have?

Reply:
> Any specific concerns that you have?

No, just wanted to ensure the new syntax has the same behavior. If you verified it, I'm fine.